### ISSUE TITLE
Better/more bumping flexibility for initial-development-versions, i.e. < 1.0.0.

### ISSUE THREAD

**Issue by rklec** [absolute-version/commit-and-tag-version#26]
It would be great to finally solve https://github.com/conventional-changelog/standard-version/issues/539

## TL;DR

In Standard-Version when you have a version < 1.0.0 it always bumps the _patch_ version, no matter what you do.
It would be great to be able to adjust this behavior.

> > [The SemVer 2.0.0 spec days](https://semver.org/spec/v2.0.0.html):
> > **How should I deal with revisions in the 0.y.z initial development phase?**
> >
> >The simplest thing to do is start your initial development release at 0.1.0 and then increment the minor version for each subsequent release.
> 
> **However**, this is only the FAQ part and a _suggestion_ (i.e. "The simplest thing to do").
> 
> What the spec really defines is:
> > 4. Major version zero (0.y.z) is for initial development. Anything MAY change at any time. The public API SHOULD NOT be considered stable.
> 
> It _explicitly_ allows y (the minor version) to change.
> 
> The thread in https://github.com/semver/semver/issues/333#issuecomment-463390138 tackles this issue and I guess this summarized the current state quite good:
> 
> > There is nothing explicit about what the rules for initial development should be.
> 
> As such, IMHO, there should at least be a flag/configuration option in `.versionrc` to _also_ bump minor in the initial development stage (<1.0.0) just like you would do when it is stable.
> Otherwise the version always stays at `0.0.z` until `1.0.0` which feels awkward to me and let me to find this issue as it was clearly unexpected/not what I thought standard-version would behave.
> 
> Both variants are clearly in line with the spec, so IMHO, standard-version should support it. You could even, and that may be discussed, doing that new behavior by default. Or better introduce it as optional for now and later make it a default, if you want to go that way.
> BTW SemVer 2.0.0 also says:
> The simplest thing to do is start your initial development release at 0.1.0 and then increment the **minor** version for each subsequent release.
>
>That's _not_ the **patch** version (0.0.z), as Standard-Version currently does! Minor would make more, sense IMHO, but still the usual bumping behavior would be better. (0.y.0)

### Idea

* The default behavior may be changed. Maybe bump the minor version? Or better use the "default behaviour" for bumping as described above.
* An entry in `.versionrc` where you can choose what to bump in pre-release versions like `minor` or `patch` (you can set it to `patch` to keep/restore the old behavior of standard-version) or `auto` (or maybe called `default`), where it chooses based on whether it's a `feat`(ure) or (bug) `fix`.

**Comment by TimothyJones** [comment]
I'm going to close this for now, please reopen if you'd like to discuss this more.

**Comment by TimothyJones** [comment]
I've added some additional discussion here to describe the fork rationale: https://github.com/absolute-version/commit-and-tag-version/discussions/30

**Comment by TimothyJones** [comment]
> are not consumed and relied on like "packages"

Hmm. I don't think semver is for this usecase, though. The spec says:

>  Software using Semantic Versioning MUST declare a public API. This API could be declared in the code itself or exist strictly in documentation. However it is done, it SHOULD be precise and comprehensive.

Anyway, could you elaborate on what you mean by "properly keep track of new features and bugfixes til then"? You should be able to track those fine, but what will happen before 1.0.0 is just that the minor version will bump for breaking changes, and the patch version will bump for everything else. As discussed above, this is by convention.

> I actually expected the `preMajor` flag to solve that.

I'm not familiar with this flag. Where did you find it documented / defined?

> Considering the laid out scenario, any chance to cover those other use-cases as well?

As above, I don't think it's appropriate to directly cover non-semver or unconventional use-cases. However, if you want to override the version number generation, you can use `--release-as` to release whatever you like.
